qiita_pet/util.py

```diff
@@ -51,8 +56,13 @@ def clean_str(item):
     -------
     str
         cleaned string
+
+    Notes
+    -----
+    This function removes colons and periods from strings, and replaces spaces
+    with underscores.
     """
-    return str(item).replace(" ", "_").replace(":", "")
+    return re.sub(clean_str_regex, "", str(item).replace(" ", "_"))
```

Comment:
Excellent! Thanks for checking it out.

On Tue, Feb 24, 2015 at 3:55 PM, Daniel McDonald <notifications@github.com>
wrote:

> In qiita_pet/util.py
> <https://github.com/biocore/qiita/pull/865#discussion_r25302076>:
>
> >      """
> > -    return str(item).replace(" ", "_").replace(":", "")
> > +    return re.sub(clean_str_regex, "", str(item).replace(" ", "_"))
>
> Likely just implements an array mapped against the ascii set or something,
> so should be extremely quick.
> … <#14bbdcd0b7d440a4_>
> On Tue, Feb 24, 2015 at 3:19 PM, Joshua Shorenstein <
> notifications@github.com> wrote: In qiita_pet/util.py <
> https://github.com/biocore/qiita/pull/865#discussion_r25299115>: > """ >
> - return str(item).replace(" ", "_").replace(":", "") > + return
> re.sub(clean_str_regex, "", str(item).replace(" ", "_")) Someone has
> apparently already benched this type of thing and translation tables are
> the winner:
> http://stackoverflow.com/questions/265960/best-way-to-strip-punctuation-from-a-string-in-python
> — Reply to this email directly or view it on GitHub <
> https://github.com/biocore/qiita/pull/865/files#r25299115>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiita/pull/865/files#r25302076>.
>